python/ql/src/experimental/typetracking/TypeTracker.qll

```diff
@@ -0,0 +1,420 @@
+/** Step Summaries and Type Tracking */
+
+private import TypeTrackerPrivate
+
+/**
+ * Any string that may appear as the name of a piece of content. This will usually include things like:
```

Comment:
I think "content" is more appropriate, precisely because it is not limited to just attributes/fields/properties. See for instance the  JavaScript modelling of [array accesses](https://github.com/github/codeql/blob/main/javascript/ql/src/semmle/javascript/Arrays.qll#L165). It's perhaps not immediate from the context, but this `loadStep` is used to create an `AdditionalTypeTrackingStep`, and in this case the "content" (or "property" in the JS libs) is `arrayElement()` which ultimately points to a [string value](https://github.com/github/codeql/blob/main/javascript/ql/src/semmle/javascript/dataflow/Configuration.qll#L797) of `$arrayElement$` -- an unlikely name for an object attribute. 
(For what I assume are legacy reasons, the JS libs call these "pseudo-properties", but rather than picking a too-specific name and then subsequently making it more generic, we have the chance to make it generic from the get-go. :slightly_smiling_face:)

Finally, the name "content" is also used in much the same way in [the shared dataflow library](https://github.com/github/codeql/blob/main/csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPublic.qll#L192), so I think it has merit to reuse this name (though if the two notions end up end up being too different, this might cause some confusion).

Reply:
I was afraid that the only reason for the word "content" was to be language agnostic . I agree with using the same terminology as the shared dataflow library and if it is truly meant to be more generic then just field/property/attribute then "content" makes sense. Thanks for the explanation.